### PR TITLE
Merge ECCO Python Adapter + Minor REST Backend Updates

### PR DESCRIPTION
**Pull request by MatthiasPreuner**
# Description

ECCO Python Adapter was added as a submodule.
Minor REST Backend Updates in accordance with current Frontend changes (port settings)

## Testing

Adapter can be tested via the gradle default test-task.
Evalution (for Thesis): extensive testing, performance measurements via the gradle repositoryTest task

## Checklist

- [x] I have added unit tests for new code
- [ ] I have adapted unit tests for changed code
- [ ] I have deleted all code that was commented out

### CODE REVIEW

File: .github/workflows/eccoBuild.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# This workflow uses actions that are not certified by GitHub.
```
Comment (meik99) [inline .github/workflows/eccoBuild.yml:1]: there is already a build-workflow for ecco, so this file can probably be removed

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "adapter/python"]
+	path = adapter/python
+	url = https://github.com/MatthiasPreuner/ecco-adapter-python.git
```
Comment (meik99) [inline .gitmodules:3]: a submodule for the isse ecco project should have its repository also inside the isse org

File: base/build.gradle [hunk 3]
```diff
@@ -21,7 +21,9 @@ dependencies {
 	testImplementation project(':ecco-adapter-file')
 	testImplementation project(':ecco-adapter-image')
 	testImplementation project(':ecco-adapter-text')
+	testImplementation project(':ecco-adapter-python')
 	testImplementation project(':ecco-adapter-java')
+//	testImplementation project(':ecco-adapter-java-patched')
```
Comment (meik99) [inline base/build.gradle:26]: remove commented lines
```suggestion
```

File: build.gradle [hunk 4]
```diff
@@ -58,7 +58,10 @@ subprojects {
 	}
 
 	configurations {
-		integrationTestCompile.extendsFrom testCompile
+		integrationTestCompile.extendsFrom implementation
+		integrationTestRuntimeOnly.extendsFrom runtimeOnly
+
+		//integrationTestCompile.extendsFrom testCompile
```
Comment (meik99) [inline build.gradle:64]:
```suggestion
```

File: gui/build.gradle [hunk 5]
```diff
@@ -9,6 +9,24 @@ application {
...
+
+//compileJava {
+//	doFirst {
+//		options.compilerArgs = [
+//				'--module-path', classpath.asPath,
+//				'--add-modules', 'javafx.controls,javafx.swing'
+//		]
+//	}
+//}
+
```
Comment (meik99) [inline gui/build.gradle:29]:
```suggestion
```

File: rest/README.md [hunk 6]
```diff
@@ -14,8 +14,8 @@ mirconaut-cli.yml contains the config for possible CLI usage.
 ## API Documentation
 
 This server provides an OpenAPI documentation as well as a Swagger UI.
-After the server has started, it is accessible via http://localhost:8080/swagger-ui.
-The raw OpenAPI documentation can be accessed via http://localhost:8080/swagger/ecco-restservice-0.0.1.yml.
+After the server has started, it is accessible via http://localhost:8081/swagger-ui.
```
Comment (meik99) [inline rest/README.md:17]: just out of curiosity, why change it to 8081?

File: rest/src/main/java/at/jku/isse/ecco/rest/Settings.java [hunk 7]
```diff
@@ -27,4 +34,12 @@ private static String getAutoLocation() {
             }
         }
     }
+
+    private static Boolean isRunningInsideDocker() {
+        try (Stream<String> stream = Files.lines(Paths.get("/proc/1/cgroup"))) {
```
Comment (meik99) [inline rest/src/main/java/at/jku/isse/ecco/rest/Settings.java:39]: this check will fail if the host is using cgroup v2.
first check if the file "/.dockerenv" exists (see https://github.com/localstack/localstack/pull/4352/files)
if that check fails the check you already have, i.e., looking for "/docker" inside the cgroup v1 path.

I would also suggest looking for a library, there might be already someone that has solved this problem

**Review by meik99 — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 2 above)
Comment (MatthiasPreuner) [reply .gitmodules:3]: should we make an adapter repository on isse org?

File: .github/workflows/eccoBuild.yml (hunk 1 above)
Comment (Dorkat0) [reply .github/workflows/eccoBuild.yml:1]: You are right, didn't we delete that already?

File: rest/README.md (hunk 6 above)
Comment (Dorkat0) [reply rest/README.md:17]: Because we change the Frontend to 8080.

**Review by meik99 — APPROVED** [review]

**Pull request merged**